lib/board.rb

```diff
@@ -0,0 +1,71 @@
+require './lib/piece'
+require './lib/null_piece'
+
+class Board
+
+  attr_reader :col_indices, :columns, :game_state
+
+  def initialize()
+    @col_indices = {
+                    "A" => 0, 
+                    "B" => 1, 
+                    "C" => 2, 
+                    "D" => 3, 
+                    "E" => 4, 
+                    "F" => 5, 
+                    "G" => 6
+                  }
+    @columns = [[],[],[],[],[],[],[]]
+    @game_state = 'none' #consider relocating depending on win_check and game class logic
+  end
+
+  def valid_player_input?(player_input)
+    @col_indices.keys.include?(player_input.upcase)
```

Comment:
Good idea, maybe since we have other validity checks we separate them from both classes and create a validitycheck class. I'll keep the code here for now and we can make a separate branch for that feature.